### PR TITLE
Fix CS7036 and CS1501 errors showing instead of CS0012

### LINKED ISSUE #5043: Roslyn may give unmeaningful error messages when calling methods uses unreferenced types
When a method in a referenced library has `params` parameters, and takes or returns unreferenced types, Roslyn will give unmeaningful error message.

Example:

```` C#
// LibraryA.cs
// referenced System.Drawing.dll
public class LibraryA
{
    public static System.Drawing.Color GetColor(params int[] args) { /* ... */ }
    // Or
    public static void SetColor(params System.Drawing.Color[] args) { /* ... */ }
}
````

````C#
// LibraryB.cs
// NOT referenced System.Drawing.dll
// referenced LibraryA.dll
public class LibraryB
{
    public static void TestMethod()
    {
        LibraryA.GetColor();
        // CS7036 There is no argument given that corresponds to the required
        // formal parameter 'args' of 'LibraryA.GetColor(params int[])'

        LibraryA.GetColor(1);
        // OK: CS0012 The type 'Color' is defined in an assembly that is not referenced.
        // You must add a reference to assembly 'System.Drawing, ...'.

        LibraryA.GetColor(1, 1);
        // CS1501 No overload for method 'GetColor' takes 2 arguments.
    }
}

### PR DESCRIPTION
**Pull request by dpen2000**
Fixes #5043

### CODE REVIEW

**Comment by dpoeschl** [comment]
retest prtest/mac/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_mac_dbg_unit32/1800/
// Retest reason: See #7272

**Comment by dpen2000** [comment]
Just fixed typo in test name (param to params) but before this, last mac build failed again: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_mac_dbg_unit32/1802/

**Comment by davkean** [comment]
tag @dotnet/roslyn-compiler

File: src/Compilers/CSharp/Test/Semantic/Semantics/UseSiteErrorTests.cs [hunk 1]
```diff
@@ -1933,6 +1933,42 @@ static void Main()
...
+        {
+            var missingSource = @"
+public class Missing { }
+";
+
+            var libSource = @"
+public class C
+{
+    public static Missing GetMissing(params int[] args) { return null; }
+    public static void SetMissing(params Missing[] args) { }
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/UseSiteErrorTests.cs]: Please add a unit-test for scenarios where params is is not the only parameter of the methods.

**Comment by dpen2000** [comment]
@AlekseyTs, I've added extra scenarios where string is the first parameter now instead of the params array. Thanks

**Comment by AlekseyTs** [comment]
LGTM

**Comment by davkean** [comment]
Mac machines were turned off during the build.

@dotnet-bot test prtest/mac/dbg/unit32 please
@dotnet-bot test eta please

**Comment by davkean** [comment]
@dotnet-bot retest prtest/mac/dbg/unit32 please

**Comment by dpen2000** [comment]
Is this okay now except for mac build, @davkean?

**Comment by davkean** [comment]
You can ignore Mac results - we've disabled them for the moment.

Can we get another member from @dotnet/roslyn-compiler to sign-off on this?

**Comment by jaredpar** [comment]
This seems to have the potential to change overload resolution results because it can add new members to the candidate set.  This makes me more nervous about this change.  Have we looked into whether this could affect overload resolution or not?

**Comment by AlekseyTs** [comment]
@jaredpar I don't think this change can have negative impact on overload resolution. Methods with bad types are never applicable, regardless whether they are considered in normal or expanded form.

**Comment by jaredpar** [comment]
@AlekseyTs okay i see a bit better now.  The return is of `true` for the error condition is guarded by the initial check which guarantees the expanded form is in an error state.  

:+1:

**Comment by davkean** [comment]
Thanks for the contribution!

**Comment by dpen2000** [comment]
Thanks!

**Pull request merged**